mlir/lib/Dialect/SCF/Utils/Utils.cpp

```diff
@@ -498,6 +498,21 @@ FailureOr<UnrolledLoopInfo> mlir::loopUnrollByFactor(
   return resultLoops;
 }
 
+/// Unrolls this loop completely.
+LogicalResult mlir::loopUnrollFull(scf::ForOp forOp) {
+  IRRewriter rewriter(forOp.getContext());
+  std::optional<uint64_t> mayBeConstantTripCount = getConstantTripCount(forOp);
+  if (mayBeConstantTripCount.has_value()) {
```

Comment:
https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code